daemon/daemon_linux.go

```diff
@@ -0,0 +1,32 @@
+//go:build !windows && !darwin
+// +build !windows,!darwin
+
+/*
+ * Copyright (c) 2022, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package daemon
+
+import (
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+func processExists(pid int) bool {
+	_, err := os.Stat(filepath.Join("/proc", strconv.Itoa(pid)))
```

Comment:
This will stop easeprobe functioning under non linux systems :cry:

Reply:
No worry. This file is only compiled for the non-windows and non-mac systems. 

notice the file name with the `_linux` suffix. and there are two comment lines at the beginning of this file to indicate the build platform.

```
//go:build !windows && !darwin
// +build !windows,!darwin
```

Reply:
oh ok actualy this is what confused me the `!windows and !darwing` excludes these, so `linux` & `openbsd` should be included when trying to build this?

Reply:
Yes, in Go land, we can run `go tool dist list` you can see all of the platform and CPU arch it supports to do the cross-compiling. (The list as below)

And we can use the `go:build` or `+build` to let go compiler understand the source file can be compiled on what platform. for the `processExists()` function, I implement 3 version for each different platform. 

```shell
$ go tool dist list                                                                                            ─╯
aix/ppc64
android/386
android/amd64
android/arm
android/arm64
darwin/amd64
darwin/arm64
dragonfly/amd64
freebsd/386
freebsd/amd64
freebsd/arm
freebsd/arm64
illumos/amd64
ios/amd64
ios/arm64
js/wasm
linux/386
linux/amd64
linux/arm
linux/arm64
linux/mips
linux/mips64
linux/mips64le
linux/mipsle
linux/ppc64
linux/ppc64le
linux/riscv64
linux/s390x
netbsd/386
netbsd/amd64
netbsd/arm
netbsd/arm64
openbsd/386
openbsd/amd64
openbsd/arm
openbsd/arm64
openbsd/mips64
plan9/386
plan9/amd64
plan9/arm
solaris/amd64
windows/386
windows/amd64
windows/arm
windows/arm64
```

Reply:
Thank you so much for the explanation :heart: